src/script/page/RightSidebar/utils/toggleRightPanel.ts

```diff
@@ -0,0 +1,190 @@
+/*
+ * Wire
+ * Copyright (C) 2022 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import React from 'react';
+import {createRoot, Root} from 'react-dom/client';
+import {MainViewModel} from '../../../view_model/MainViewModel';
+import {rightPanelAnimationTimeout} from '../RightSidebar';
+
+let elementContainer: HTMLDivElement | undefined;
+let reactRoot: Root;
+
+const elementId = 'app';
+
+export const forceCloseRightPanel = () => {
+  if (elementContainer) {
+    reactRoot.unmount();
+    document.getElementById(elementId)?.removeChild(elementContainer);
+    elementContainer = undefined;
+  }
+};
+
+const applyStyle = (element: HTMLElement, style: Record<string, string>) => {
+  if (element) {
+    Object.entries(style).forEach(([key, styleValue]) => (element.style[key as any] = styleValue));
+  }
+};
+
+const commonPanelStyle = {
+  position: 'absolute',
+  right: '0',
+  width: `${MainViewModel.CONFIG.PANEL.WIDTH}px`,
+};
+
+const panelStyle = {
+  closed: {
+    ...commonPanelStyle,
+    transform: `translateX(${MainViewModel.CONFIG.PANEL.WIDTH}px)`,
+  },
+  open: {
+    ...commonPanelStyle,
+    transform: 'translateX(0px)',
+  },
+};
+
+interface RenderElement {
+  onClose?: () => void;
+}
+
+const toggleRightSidebar =
+  <T extends RenderElement>(Component: React.FC<T>) =>
+  (props: T) => {
+    const rightPanelId = 'right-column';
+    const rightPanelClassName = 'right-column';
+    const widthTransition = `width ${rightPanelAnimationTimeout}ms cubic-bezier(0.19, 1, 0.22, 1)`;
+
+    const app = document.querySelector<HTMLElement>('#app');
+    const titleBar = document.querySelector<HTMLElement>('#conversation-title-bar');
+    const input = document.querySelector<HTMLElement>('#conversation-input-bar');
+    const rightPanel = document.querySelector<HTMLElement>('#right-column');
+
```

Comment:
Wow this is scary. Isn't there a way to do this without querying the DOM? 
This is not going to go well if we change the ids/classes.